experimental/adaptive-dialog/docs/anatomy-and-runtime-behavior.md

```diff
@@ -0,0 +1,82 @@
+
+### Rules
+_Rules_ enable you to catch and respond to events. The broadest rule is the EventRule that allows you to catch and attach a set of steps to execute when a specific event is emitted by any sub-system. Adaptive dialogs supports couple of other specialized rules to wrap common events that your bot would handle. See [here][2] to learn more about supported Rules and their usage.
+
+### Steps
+_Steps_ help put together the flow of conversation when a specific event is captured via a Rule. **_Note:_** unlike Waterfall dialog where each step is a function, each step in an Adaptive dialog is in itself a dialog. This enables adaptive dialogs by design to 
+- have a much cleaner ability to handle and deal with interruptions.
+- branch conditionally based on context or state.
+See [here][3] to learn more about supported steps and their usage.
+
+### Inputs
+_Inputs_ are wrappers around [prompts][2] that you can use in an adaptive dialog step to ask and collect a piece of input from user, validate and accept it into memory. Inputs include these pre-built features - 
+- Accepts a property to bind to off the new V3 style memory scope in V4. 
+- Performs existential check before prompting. 
+- Grounds input to the specified property if the input from user matches the type of entity expected. 
+- Accepts constraints - min, max, etc. 
+
+See [here][4] to learn more about supported Inputs and their usage.
+
+## Runtime behavior: Adaptive dialog
+
+To help illustrate this, let's take a scenario based walk through of the runtime behavior of Adaptive dialogs.
+
+***Travel agent bot***
+```
+    User: I’d like to book a flight
+    Bot:  Sure. What is your destination city?
+    User: How’s the weather in Seattle?
+    Bot:  Its 72 and sunny in Seattle
+    ...
+    ...
+```
+
+<p align="center">
+    <img alt="Adaptive_dialog_scenario_setup" src="./Assets/adaptive-dialog-scenario-setup.png" style="max-width:700px;" />
+</p>
+
+For this scenario, we have three adaptive dialogs - 
+1. rootDialog which is an adaptive dialog with its own 'LUIS' model and a set of rules and steps
+2. bookFlightDialog which is an adaptive dialog that can handle conversations about booking a flight
+3. weatherDialog which is an adaptive dialog that can handle conversations about getting weather information. 
+
+Here's the flow when user says `I'd like to book a flight`
+
+<p align="center">
+    <img alt="Adaptive_dialog_scenario_setup" src="./Assets/adaptive-dialog-first-utterance.png" style="max-width:700px;" />
+</p>
+
+And here's the flow when user says `How's the weather in Seattle?`
+
+<p align="center">
+    <img alt="Adaptive_dialog_scenario_setup" src="./Assets/adaptive-dialog-second-utterance.png" style="max-width:700px;" />
```

Comment:
Need to add step 10 to the diagram to show the conversation returns to the which city dialog.